Frontend/research/src/Reducer/Reducer.js

```diff
@@ -119,19 +145,36 @@ const reducer = (state = initialState, action) => {
         case DELETE_LINK:
             return {
                 ...state,
-                deletingLink: true
+                deletingLink: true,
+                // links: state.links.filter( (link) => link.id !== action.payload.id )
             }
         case DELETE_LINK_SUCCESS:
             return {
                 ...state,
                 deletingLink: false,
-                links: ''
+                links: state.links.filter((link) => link.id !== action.payload.id)
             }
         case DELETE_LINK_FAILURE:
             return {
                 ...state,
                 error: action.payload
             }
+        case UPDATE_LINK:
+            return {
+                ...state,
+                updatingLink: true,
+            }
+        case UPDATE_LINK_PASS:
+            return {
+                ...state,
+                updatingLink: false,
+                // links: state.links.filter((link) => link.id !== action.payload.id)
+            }
+        case UPDATE_LINK_FAIL:
+            return {
+                ...state,
+                error: action.payload
+            }
         default:
```

Comment:
you  can just do default: return state. the default case is the only place we dont need to spread state